alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -164,9 +164,10 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
                     matchedQueriesForDocs.forEach { hit ->
                         val (id, query) = Pair(
                             hit.id.replace("_${indexName}_${monitor.id}", ""),
-                            ((hit.sourceAsMap["query"] as HashMap<*, *>)["query_string"] as HashMap<*, *>)["query"]
+                            ((hit.sourceAsMap["query"] as HashMap<*, *>)["query_string"] as HashMap<*, *>)["query"].toString()
+                                .replace("_${indexName}_${monitor.id}", "")
```

Comment:
Why do we need indexName in the first place. Shouldn't just the ``monitor.id`` suffice to filter the query applicable for the monitor from the query index.

Reply:
The index is needed since a monitor could be monitoring multiple indices through an index pattern or an alias